### ISSUE TITLE
RCTBridge required dispatch_sync to load RCTDevLoadingView. This may lead to deadlocks

### ISSUE THREAD

**Issue by articolo** [facebook/react-native#23833]
## 🐛 Bug Report
When *FIRST* installing the app on the emulator the app hang for couple of minutes and the described warning shows and the memory usage of the app going to 10GB. Sometimes it never getting out of the "freeze" but when it does, I a have one tab that when I tap on it, it causing it to freeze again (the rest of the tabs are ok). When it stops freezing the app work again. Then when trying to close the app and open it again it works fine. When uninstalling it and reinstalling it happens. 

The bigger issue with that is when trying to run it on real device, it hangs for some time on the device and then the freezing happens again when tapping on this tab. My guess is the warning is indeed correct and something is causing a deadlock when first running the app. And I have no idea what causing it
 
the issue is going away ONLY if I close and open the app on the emulator. (Never in real device)

after its unfreeze this is how my debugger looks like : 

<img width="275" alt="Screen Shot 2019-03-09 at 23 34 35" src="https://user-images.githubusercontent.com/48390732/54077674-78401680-42c4-11e9-9df0-9ff65be9b474.png">

## To Reproduce

- uninstalling app from emulator
- reinstalling app on emulator
- waiting it to unfreeze 
- press the tab that causing it to freeze again

## Expected Behavior

the app shouldnt freeze when installing the app and when pressing the tab

## Environment

  React Native Environment Info:
    System:
      OS: macOS 10.14.3
      CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
      Memory: 142.31 MB / 16.00 GB
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 8.9.4 - /usr/local/bin/node
      Yarn: 1.5.1 - /usr/local/bin/yarn
      npm: 4.6.1 - /usr/local/bin/npm
    SDKs:
      iOS SDK:
        Platforms: iOS 12.1, macOS 10.14, tvOS 12.1, watchOS 5.1
    IDEs:
      Android Studio: 3.3 AI-182.5107.16.33.5264788
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.6.3 => 16.6.3 
      react-native: 0.57.8 => 0.57.8 
    npmGlobalPackages:
      create-react-native-app: 1.0.0
      react-native-cli: 2.0.1
      react-native-git-upgrade: 0.2.7
      react-native-init: 0.6.7
      react-native-rename: 2.2.2

**Comment by articolo** [comment]
After A LOT of testing and investigations I found the problem - it's the ` this.setState({ alertRef: this.alertRef });` in `componentdidMount` fir some reason React doesnt like changing the state here and I tried also to change setState with Redux reducer instead but its the same issue. Basically I can't change the state in `App.js` and I have no idea why. If someone have any hypothesis it will be great. But indeed after removing this call it works just fine.
